### ISSUE TITLE
TypeError: register() got an unexpected keyword argument 'base_name'

### ISSUE THREAD

**Issue by zvyn** [drf-forms/drf-schema-adapter#57]
Currently running ./test.sh fails with the following error:

```python
TypeError: register() got an unexpected keyword argument 'base_name'
```

The reason is [this change](https://github.com/encode/django-rest-framework/pull/5990) in the Django REST Framework released in version 3.9.0: "Deprecate the Router.register base_name argument in favor of basename" -- [Release Notes](https://www.django-rest-framework.org/community/release-notes/#390)

I'd propose the following fix for now:
- allow both, `base_name` and `basename` for drf-schema-adapter routers
- translate `base_name` to `basename` when passing kwargs to DRF
- throw a `ValueError` when both `base_name` and `basename` are given

If that sounds good to you I can try to come up with a patch!

**Comment by matiassimon** [comment]
I am having the same error trying to follow [this cookbook](https://drf-schema-adapter.readthedocs.io/en/latest/cookbooks/basics/endpoints/) with Python 3.7, Django 3.0 and DRF 3.11. I dont understand why the [compatibility matrix](https://github.com/drf-forms/drf-schema-adapter) says that DRF 3.10 is ok when [this change](https://github.com/encode/django-rest-framework/pull/5990) is in DRF 3.9.

**Comment by nanuxbe** [comment]
As far as I can tell, this change appeared in the documentation of DRF 3.9 but was effective only starting 3.10

Anyway, this has been fixed in release 2.0.7 (https://github.com/drf-forms/drf-schema-adapter/commit/a6c7bc07c62d96fa6680a009f79b9af8359bb93b)

Thanks both of you for the feedback

**Comment by B3ns44d** [comment]
From the  [release notes of Django RestFramework](https://www.django-rest-framework.org/community/release-notes/#390)  and  [`DRF 3.9`  announcement](https://www.django-rest-framework.org/community/3.9-announcement/#router-argumentmethod-renamed-to-use-basename-for-consistency)  they mentioned that

> Deprecate the  `Router.register`  **`base_name`**  argument in favor of  **`basename`**.  [#5990](https://github.com/encode/django-rest-framework/issues/5990)

  
Which means, the argument  **`base_name`**  is no longer available from  **`DRF=3.11`**  onwards and use  [**`basename`**](https://www.django-rest-framework.org/api-guide/routers)  instead

So, Change your router config as,

```
router.register(r'musician', MusicianViewset, basename='musician')
router.register(r'album', AlbumViewset, basename='album')
```

**Comment by longhakly** [comment]
Thank you, it works now
